### PR TITLE
Feature/orochi network framework v2

### PR DESCRIPTION
**Pull request by chiro-hiro**

### CODE REVIEW

File: packages/framework/src/model/model-mysql-basic.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  public async update(data: Partial<T>, conditions?: IModelCondition<T>[]): Promise<DBResult<T[]>> {
+    return DBResult.wrap<T[]>(
+      await this.attachConditions(this.getDefaultKnex().update(data).returning('*'), conditions)
+    );
+  }
+
+  public async forceUpdate(data: Partial<T>, conditions?: IModelCondition<T>[]): Promise<DBResult<T[]>> {
+    const [record] = await this.attachConditions(this.basicQuery(), conditions).limit(1);
+    if (typeof record === 'undefined' || typeof conditions === 'undefined') {
+      return this.insert([data]);
```
Comment (dqtkien) [inline packages/framework/src/model/model-mysql-basic.ts:64]: .returning('*')

File: packages/framework/src/model/model-mysql-basic.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    } else {
+      return this.update(data, conditions);
```
Comment (dqtkien) [inline packages/framework/src/model/model-mysql-basic.ts:66]: `this.update(data, conditions).returning('*');`

File: packages/framework/src/interface/common.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+export interface IModelCondition<T> {
+  field: keyof T;
+  operator?: '>' | '<' | '<=' | '>=' | '=' | '!=';
+  value: string | number | boolean;
```
Comment (dqtkien) [inline packages/framework/src/interface/common.ts:4]: `value: string | number | boolean | bigint;`

File: packages/framework/src/model/condition.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  private currentField: keyof T = '' as keyof T;
+
+  public static getInstance<T>() {
+    return new ConditionBuilder<T>();
+  }
+
+  private newCondition(
+    field: keyof T,
+    operator: '>' | '<' | '<=' | '>=' | '=' | '!=',
+    value: string | number | boolean
```
Comment (dqtkien) [inline packages/framework/src/model/condition.ts:15]: Do we need `bigint` here?

File: packages/framework/src/model/model-mysql-basic.ts (hunk 1 above)
Comment (chiro-hiro) [reply packages/framework/src/model/model-mysql-basic.ts:64]: Đoạn này nó đã returning rồi em, nên cái của em bị thừa:
https://github.com/orochi-network/one-piece/pull/36/files/c4fe5756f1a1538991d0b03fc4b361a4c13d6fd2#diff-406f9be1f0c63b821c1c3d6fc73d3f4fea4076a49b540c77c62fb6f8873cdd64R51-R53

File: packages/framework/src/model/model-mysql-basic.ts (hunk 2 above)
Comment (chiro-hiro) [reply packages/framework/src/model/model-mysql-basic.ts:66]: Already implemented here https://github.com/orochi-network/one-piece/pull/36/files/c4fe5756f1a1538991d0b03fc4b361a4c13d6fd2#diff-406f9be1f0c63b821c1c3d6fc73d3f4fea4076a49b540c77c62fb6f8873cdd64R55-R59

File: packages/framework/src/interface/common.ts (hunk 3 above)
Comment (chiro-hiro) [reply packages/framework/src/interface/common.ts:4]: For postgres `bigint` will be treat as `string`, so the type of `T` should be:

```ts
{
   bigintValue: string,
}
```

So `bigint` will create overhead for type cast

File: packages/framework/src/model/condition.ts (hunk 4 above)
Comment (chiro-hiro) [reply packages/framework/src/model/condition.ts:15]: No

**Review by dqtkien — APPROVED** [review]

**Pull request merged**